src/utils/Redis.ts

```diff
@@ -0,0 +1,49 @@
+import { Either, left, right } from "fp-ts/lib/Either";
+import { fromNullable, Option } from "fp-ts/lib/Option";
+import { RedisClient } from "redis";
+
+export function redisSet(
+  redisClient: RedisClient,
+  key: string,
+  value: string,
+  mode: string,
+  duration: number
+): Promise<Either<Error, "OK" | undefined>> {
+  return new Promise(resolve => {
+    if (!redisClient.connected) {
```

Comment:
isn't there an autoconnect option? @lussoluca how do we do this in the backend? we should share this logic and avoid resolving the problem twice

Reply:
we don't. I've asked @lussoluca to create a story about it